### ISSUE TITLE
Custom variable resolution happens before properties it depends on are resolved

### ISSUE THREAD

**Issue by williamsandonz** [serverless/serverless#7867]
When my local plugin's variable resolver method is called, upon accessing variables defined in this.serverless.service , any values with ${self:foo} are not interpolated making them unusable. 

<details>
<summary><code>serverless.yml</code></summary>

```yaml
app: my-app
service: my-service
plugins:
  - my-plugin
provider:
  name: aws
  runtime: nodejs12.x
  stackName: ${self:app}-${self:provider.stage}-${self:service}
  stage: dev
functions:
  hello:
    handler: handler.hello
resources:
  Resources:
    Bucket:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: ${foo:bar}
```
</details>

And if you create a local plugin at .serverless_plugins / my-plugin

<details>
<summary><code>Plugin</code></summary>

```javascript
'use strict';

class MyPlugin {
  constructor(serverless, options) {
    this.serverless = serverless;
    this.options = options;
    this.variableResolvers = {
      foo: {
        isDisabledAtPrepopulation: true,
        resolver: (path) => {
          console.log(this.serverless.service.provider.stackName);
        },
        serviceName: 'Not prepopulated...',
      }
    };
  }
}
module.exports = MyPlugin;
```
</details>

<details>
<summary><b><code>sls deploy</code> output</b></summary>

```
${self:app}-${self:provider.stage}-${self:service}
```
But expected 

```
my-app-dev-my-service
```
</details>

<b>Installed version</b>

```
Framework Core: 1.73.1
Plugin: 3.6.13
SDK: 2.3.1
Components: 2.31.2
```
This is a problem in my situation because my variable resolution requires using aws-sdk to call describe-stacks which requires accessing this.serverless.service.provider.stackName and  it's quite common for this to be defined with ${self:app}-${self:provider.stage}-${self:service}.

**Comment by medikoo** [comment]
@williamsandonz You're using outdated version of a framework, please upgrade

**Comment by williamsandonz** [comment]
Thanks @medikoo I've updated to the latest version and can confirm is issue is still occurring. I've also updated my 'installed version' in above post.

**Comment by medikoo** [comment]
@williamgueiros I've checked that and it works like that, because it's not evident for internals that resolution of `foo` depends on `provider.stackName` being resolved first.

There's a _prepoluation_ stage which is supposed to guarantee that certain _core_ properties are resolved before other variables resolution takes place, but `provider.stackName` is not considered as _core_ property (and that seems right).

I think solution for that problem, would be to introduce a way to mark a dependent properties for custom variable sources. e.g. when defining a `foo` resolver you may also configure a paths for properties that should be resolved before attempting to resolve `foo` scope. I've added a note about such requirement to #7207 where we discuss refactor of variable resolution logic.

**Comment by williamsandonz** [comment]
Can confirm this has been fixed by 2.27.0 ([here](https://github.com/serverless/serverless/pull/8987/files)) Thank you @medikoo !
